### DISCUSSION TITLE
product filtering and attribute doesn't have important function ?

### DISCUSSION THREAD

**Discussion by matteobaro** [saleor/saleor#8806] [Q&A]
Hello,
I'm new with saleor and i'm trying to build a new ecommerce with it.
I have encountered a problem with building sidebar with attributes filter...

i need to show only attribute filter values of the product actually displayed in page

example to be more clear:
i make a search on the ecommerce: 
i search trouser
in the attribute size i want to see only sizes of trouser, not sizes of other clothing type

i have already searched in the documentation but i didn't find product count for attribute, or a graphql with list of attribute actually available in the page...

it is possibile to do that with saleor?

I hope I explained myself

Thanks

**Comment by ciocca-dev** [comment]
I'm interested too, i think this is a very important feature

**Comment by maarcingebala** [comment]
Maybe you should try to define separate attributes for each type of product, e.g. if you have a product type "Trousers", then you should also have an attribute "trousers-size". This way you should be able to only get the filters from this particular product type in your storefront. It seems that now you have a generic attribute that is shared between different product types or categories.

**Reply by matteobaro** [reply]
I'm sorry but it's a pretty useless solution, especially with an ecommerce with many categories such as a fashion ecommerce, with many different categories displayed on the same page. With a large catalog it becomes uncomfortable and unmanageable ...
Incredible that saleor does not have this fundamental function for an ecommerce ...

vendure example:

{
    query SearchFacets($collectionSlug: String, $searchText: String, $facetValueFilters: [FacetValueFilterInput!]) {
  search(
    input: {collectionSlug: $collectionSlug, term: $searchText, facetValueFilters: $facetValueFilters, groupByProduct: true}
  ) {
    facetValues {
      facetValue {
        id
        name
        facet {
          code
        }
      }
    }
  }
}

**Reply by karolkielecki** [reply]
Thanks @matteobaro. We'll consider your feedback. Some of our clients use dedicated tools such as Algolia.

**Comment by DeanDeBlock** [comment]
hey guys, I am also playing around with Saleor and was looking for this feature.

Has this been added in the meantime?

My guess there should be a filter object returned in the products call where all values of the filters are only a reflection of the current products search


I am also building a storefront for a client in the fashion industry and without this it would not fit our needs
